query-connector/src/app/ui/designSystem/timeboxing/DateRangePicker.tsx

```diff
@@ -0,0 +1,263 @@
+      errors.end = "End date cannot be before start date.";
+    }
+
+    document
+      .getElementById("log-date-start")
+      ?.classList.toggle(styles.bgErrorLighter, !!errors.start);
+    document
+      .getElementById("log-date-start")
+      ?.classList.toggle(styles.borderError, !!errors.start);
+    document
+      .getElementById("log-date-end")
+      ?.classList.toggle(styles.bgErrorLighter, !!errors.end);
+    document
+      .getElementById("log-date-end")
+      ?.classList.toggle(styles.borderError, !!errors.end);
+
+    setDateErrors(errors);
+    return errors;
+  };
+
+  useEffect(() => {
+    setStartDate(initialStart);
+    setEndDate(initialEnd);
+  }, [initialStart, initialEnd]);
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        containerRef.current &&
+        !containerRef.current.contains(event.target as Node)
+      ) {
+        setIsOpen(false);
+      }
+    };
+
+    if (isOpen) {
+      document.addEventListener("mousedown", handleClickOutside);
+    } else {
+      document.removeEventListener("mousedown", handleClickOutside);
+    }
+
+    return () => document.removeEventListener("mousedown", handleClickOutside);
+  }, [isOpen]);
+
+  useEffect(() => {
+    if (isOpen) {
+      const startInput = document.getElementById(
+        "log-date-start",
+      ) as HTMLInputElement;
+      const endInput = document.getElementById(
+        "log-date-end",
+      ) as HTMLInputElement;
+      if (startInput)
+        startInput.value = startDate ? startDate.toLocaleDateString() : "";
+      if (endInput)
+        endInput.value = endDate ? endDate.toLocaleDateString() : "";
+    }
+  }, [isOpen, startDate, endDate]);
+
+  return (
+    <div className={styles.datePickerContainer} ref={containerRef}>
+      <div className={styles.textInputContainer}>
+        <TextInput
+          type="text"
+          id="date-range-input"
+          name="date-range-input"
+          aria-label="Date range input"
+          value={displayText}
+          readOnly
+          onClick={() => setIsOpen((prev) => !prev)}
+        />
+        <Icon.CalendarToday
+          size={3}
+          className={styles.calendarIcon}
+          aria-label="date-range-calendar-icon"
+        />
+      </div>
+      {isOpen && (
+        <div className={styles.customDateRangeWrapper}>
+          <div className={styles.datePickerModal}>
+            <div className={styles.clearButtonContainer}>
+              <Button
+                unstyled
+                type="button"
+                data-testid="date-range-clear-button"
+                className={styles.clearButton}
+                onClick={() => {
+                  setStartDate(null);
+                  setEndDate(null);
+                  setDateErrors({});
+                  setIsOpen(false);
+                  onChange({ startDate: null, endDate: null });
+                  setTimeout(() => {
```

Comment:
is there a reason we're using set timeout here to trigger these side effects? especially since we're not passing in a delay, could we just do it in the body of the onclick?